release/src/Util.hs

```diff
@@ -254,6 +259,26 @@ artifactFiles allArtifacts art@Artifact{..} = do
         [(directory </> scalaSourceJarIn, outDir </> scalaSourceJarOut) | shouldRelease allArtifacts (PlatformDependent False), Just scalaSourceJarIn <- pure mbScalaSourceJarIn] <>
         [(directory </> javadocJarIn, outDir </> javadocJarOut) | shouldRelease allArtifacts (PlatformDependent False), Just javadocJarIn <- pure mbJavadocJarIn]
 
+-- | Given an artifact, produce a list of pairs of an input file and the Maven coordinates
+artifactCoords :: E.MonadThrow m => AllArtifacts -> Artifact PomData -> m [(MavenCoords, Path Rel File)]
+artifactCoords allArtifacts art@Artifact{..} = do
+    let PomData{..} = artMetadata
+    let (directory, name) = splitBazelTarget artTarget
+    let jarClassifier =  if getPlatformDependent artPlatformDependent then Just osName else Nothing
+    directory <- parseRelDir $ unpack directory
+
+    mainArtifactFile <- parseRelFile $ unpack $ mainFileName artReleaseType name
+    pomFile <- parseRelFile (unpack (name <> "_pom.xml"))
+    mbSourceJarFile <- traverse (parseRelFile . unpack) (sourceJarName art)
+    mbScalaSourceJarFile <- traverse (parseRelFile . unpack) (scalaSourceJarName art)
+    mbJavadocJarFile <- traverse (parseRelFile . unpack) (javadocJarName art)
+
+    pure $ [ (MavenCoords { groupId = pomGroupId, artifactId = pomArtifactId, version = pomVersion, classifier = jarClassifier, artifactType = mainExt artReleaseType }, directory </> mainArtifactFile) | shouldRelease allArtifacts artPlatformDependent] <>
+           [ (MavenCoords { groupId = pomGroupId, artifactId = pomArtifactId, version = pomVersion, classifier = Nothing, artifactType = T.pack "pom" }, directory </> pomFile) | isJar artReleaseType, shouldRelease allArtifacts (PlatformDependent False)] <>
+           [ (MavenCoords { groupId = pomGroupId, artifactId = pomArtifactId, version = pomVersion, classifier = Just $ T.pack "sources", artifactType = T.pack "jar" }, directory </> sourceJarFile) | shouldRelease allArtifacts (PlatformDependent False), Just sourceJarFile <- pure mbSourceJarFile] <>
+           [ (MavenCoords { groupId = pomGroupId, artifactId = pomArtifactId, version = pomVersion, classifier = Just $ T.pack "sources", artifactType = T.pack "jar" }, directory </> scalaSourceJarFile) | shouldRelease allArtifacts (PlatformDependent False), Just scalaSourceJarFile <- pure mbScalaSourceJarFile] <>
+           [ (MavenCoords { groupId = pomGroupId, artifactId = pomArtifactId, version = pomVersion, classifier = Just $ T.pack "javadocs", artifactType = T.pack "jar" }, directory </> javadocSourceJarFile) | shouldRelease allArtifacts (PlatformDependent False), Just javadocSourceJarFile <- pure mbJavadocJarFile]
```

Comment:
Instead of creating 5 singleton lists and concatenating them, could you please create just one list with 5 elements.

Reply:
I think that this not possible due to guards each of the elements.

Reply:
Yeah, you're right. I didn't scroll all the way to the end. My bad. :|